### PR TITLE
Fix model_cropper not resetting image.num_points3D of cropped_rec

### PR DESCRIPTION
**Pull request by ArneSchulzTUBS**
While investigating the model_splitter issue for the current revision 1f95f3ee74fdbc020ef2dca190ec56958ea5f9e1 described in https://github.com/colmap/colmap/issues/2096 I found a code regression introduced with commit https://github.com/colmap/colmap/commit/67029ad21205fac3d149e06000c1e20bf4be1b80

This affects model_cropper and model_splitter tools.

As the previous implementation used the Image::ResetPoint3DForPoint2D method inside the loop, the num_points3D member was reset to 0 correctly.

https://github.com/colmap/colmap/blob/1f95f3ee74fdbc020ef2dca190ec56958ea5f9e1/src/colmap/scene/image.cc#L97-L103

When directly assigning the kInvalidPoint3DId value, new_image keeps the value of the source num_points3D representing an incorrect state of the newly created cropped_reconstruction and finally leads to a failing check at https://github.com/colmap/colmap/blob/1f95f3ee74fdbc020ef2dca190ec56958ea5f9e1/src/colmap/scene/reconstruction.cc#L184

Fixing this by adding the call to Image::ResetPoint3DForPoint2D inside the loop again as it was first implemented in https://github.com/colmap/colmap/commit/99c7904c4a3c99c17d80fee1d8b57ed7fe5d3753.

<details>
<summary>Debugger screenshots:</summary>

Current Revision 1f95f3ee74fdbc020ef2dca190ec56958ea5f9e1 :

colmap_crop_debug_faulty

Old implementation:
colmap_crop_debug_pre_change

</details>

### CODE REVIEW

File: src/colmap/scene/reconstruction.cc [hunk 1]
```diff
@@ -452,8 +452,8 @@ Reconstruction Reconstruction::Crop(
   for (const auto& image : images_) {
     auto new_image = image.second;
     new_image.SetRegistered(false);
-    for (auto& point2D : new_image.Points2D()) {
-      point2D.point3D_id = kInvalidPoint3DId;
+    for (point2D_t pid = 0; pid < new_image.NumPoints2D(); ++pid) {
```
Comment (ahojnnes) [inline src/colmap/scene/reconstruction.cc:455]:
```suggestion
    const num_points2D = new_image.NumPoints2D();
    for (point2D_t point2D_idx = 0; point2D_idx < num_points2D; ++point2D_idx) {
```

File: src/colmap/scene/reconstruction.cc [hunk 2]
```diff
@@ -452,8 +452,8 @@ Reconstruction Reconstruction::Crop(
...
+      new_image.ResetPoint3DForPoint2D(pid);
```
Comment (ahojnnes) [inline src/colmap/scene/reconstruction.cc:456]:
```suggestion
      new_image.ResetPoint3DForPoint2D(p oint2D_idx);
```

**Review by ahojnnes — CHANGES_REQUESTED** [review]
Thanks for the fix. Small suggestions for consistency in the code base.

File: src/colmap/scene/reconstruction.cc (hunk 1 above)
Comment (ArneSchulzTUBS) [reply src/colmap/scene/reconstruction.cc]: Added with type specifier in https://github.com/colmap/colmap/pull/2557/commits/6f716b8cdc107e076313934578a8a3eeeeab5ee3

File: src/colmap/scene/reconstruction.cc (hunk 2 above)
Comment (ArneSchulzTUBS) [reply src/colmap/scene/reconstruction.cc]: Changed with remove white space in https://github.com/colmap/colmap/pull/2557/commits/6f716b8cdc107e076313934578a8a3eeeeab5ee3

**Review by ahojnnes — APPROVED** [review]

**Pull request merged**